public_html/wp-content/mu-plugins/blocks/assets/src/speakers/block-content.js

```diff
@@ -19,23 +19,39 @@ import AvatarImage from '../shared/avatar';
 import './block-content.scss';
 
 class SpeakersBlockContent extends Component {
+	// it's not immediately obvious why `add` is here, so jsdoc would really help explain that. we might not want to keep it, though (see below)
 	static maybeAddMoreLink( content, add ) {
 		if ( add ) {
+			/*
+			 this feels a bit odd. is it here because JSX can only contain expressions, so we cant do something like:
+
+			if ( speaker_link ) { addMoreLink() )
+
+			what if instead we used a ternary?
+
+			{ ( speaker_link ? this.constructor.addMoreLink( post.excerpt.rendered ) : post.excerpt.rendered ).trim() }
+
+			then this function would be more straightforward, and the calling function read more naturally and be self-explanatory
+			*/
+
 			const more = sprintf(
 				'<a href="#" class="wordcamp-speaker-more-link">%s</a>',
+				// should the href have the actual value? blocks like Recent Posts have real links, but make them open in a new window if clicked on inside gutenberg
 				__( 'Read more', 'wordcamporg' )
 			);
 
 			const pattern = /<\/p>$/;
 
 			if ( Array.isArray( content.match( pattern ) ) ) {
 				content = content.replace( pattern, ' ' + more + '</p>' );
+				// could post.content.rendered contain divs or other things we need to worry about here?
 			} else {
 				content += ' ' + more;
 			}
 		}
 
 		return content;
+		// if we return content.trim() then we wouldn't have to worry about using it in the caller statements
```

Comment:
All of the above are good points, and I agree that the whole `maybeAddMoreLink` method feels a bit hacky (along with the PHP version). The problem is that I couldn't find a good way to consistently get the Read More link inside the rendered content, on the same line as the trimmed excerpt (instead of in a new, separate paragraph). For the record, I have strong negative feelings about the way excerpts are implemented in WP ;P

I would love to get some collaboration on this to figure out how to do this in a way in which we don't need this method at all :D  At the very least, I agree with your suggestions, and we should implement them.